brave/brave-ios
# Fix #5465: Wallet ramp provider

## Summary of Changes
New flow for user purchasing crypto tokens via buy with providers:
1. ramp.network
2. sardine
3. wyre
Other than buy token and buy amount, user now can choose one of the available currency. 


This pull request fixes #5465 #6032

## Submitter Checklist:


## Test Plan:
1. Check testnets do not support buy
2. Compare with desktop to see if available tokens for buy in each supported networks are correct 
3. Compare with desktop to see available providers for buy a specific token are correct.
4. make sure user selected token/ buy currency/buy amount reflects to the provider website correctly. (Update: Currency selection will be applied as a query value for the provider url. However, it depends on how the provider to will handle it. For example, Rmap.Network will check your IP address to determine the currency value regardless what query value we passed.)

1. Change your device preferred language to be not `en-us`, check Sardine does not show up in any network or tokens.

## Screenshots:


![Simulator Screen Shot - iPhone 14 Pro - 2022-10-17 at 23 33 48](https://user-images.githubusercontent.com/1187676/196330410-cd7f92c2-7991-4401-a39b-7a39a1ea61e1.png) | ![Simulator Screen Shot - iPhone 14 Pro - 2022-10-17 at 23 33 36](https://user-images.githubusercontent.com/1187676/196330422-fbb7769c-2f0c-4cbf-b969-09c7fe1bac36.png)
--|--

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
